### PR TITLE
Update build-tools to 0.24.0

### LINKED ISSUE #807: Tracking semconv build-tools 0.24.0 release readiness and roll out plan
The next version of build-tools will contain the following new features:
- [BREAKING] Requires explicit stability on each metric, attribute, and enum member and renders it on everything
  - [x] https://github.com/open-telemetry/build-tools/pull/267
  - [x] https://github.com/open-telemetry/build-tools/pull/272
  - [x] https://github.com/open-telemetry/build-tools/pull/244
  - [x] https://github.com/open-telemetry/build-tools/pull/278
- [x] Add backward compatibility check - https://github.com/open-telemetry/build-tools/pull/271
- [x] Code generation improvements - https://github.com/open-telemetry/build-tools/pull/295
- [x] Support lenient yaml validation for code generation and back-compat checks  - https://github.com/open-telemetry/build-tools/pull/300
- various cosmetic improvements

Here're the work items we need to complete inside semconv repo prior to updating build-tools version:
- [x] https://github.com/open-telemetry/semantic-conventions/pull/801
- [x] https://github.com/open-telemetry/semantic-conventions/pull/819

Update to 0.24.0 PR and fix new violations
- [ ] https://github.com/open-telemetry/semantic-conventions/pull/820

After update:
- [ ] https://github.com/open-telemetry/semantic-conventions/pull/764
- Unblock PRs blocked on resource/span attributes:
   - [ ]  https://github.com/open-telemetry/semantic-conventions/pull/681
   - [ ]  https://github.com/open-telemetry/semantic-conventions/pull/758
- Language SIGs can start using new code-generation

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #807

**For the purpose of code review: Please review YAML changes - markdown files are autogenerated**

## Changes

- update build-tools version to the next one (0.24.0 unreleased)
- Adds stability to enum members
- Add stability to a few experimental + deprecated attributes 

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

**Comment by lmolkova** [comment]
Dear codeowners, please review md and yaml files in your area

Changes to pay special attention to:
- all auto-generate attributes and metrics tables should have `Stability` column with a badge - make sure it matches your expectations.

Cosmetic changes:
- Ordering now take requirement level into consideration
- Requirement Level is rendered slightly differently than before

**Comment by joaopgrassi** [comment]
Can we run the tools and tackle each change in a separated PR? For example the sorting now changes a lot of things and it makes it hard to see what actually is a change vs what is just "cosmetic". I think that will make the review process easier.

**Comment by lmolkova** [comment]
> Can we run the tools and tackle each change in a separated PR? For example the sorting now changes a lot of things and it makes it hard to see what actually is a change vs what is just "cosmetic". I think that will make the review process easier.

Great idea, but it'd only be possible if I manually handcrafted each change. We wouldn't be able to merge them because the md check would fail.
I think going forward we should release build-tools more  (maybe each substantial md change) so that we don't have enormous diffs.

File: model/resource/telemetry.yaml [hunk 1]
```diff
@@ -25,28 +25,40 @@ groups:
           members:
             - id: cpp
               value: "cpp"
+              stability: experimental
```
Comment (lmolkova) [inline model/resource/telemetry.yaml:28]: TODO: most of these values should be stable, checking.

Comment (lmolkova) [reply model/resource/telemetry.yaml]: Updated based on the stability of the corresponding OpenTelemetry SDK.
All OTel SDKs set the attribute, and only rust is not stable yet, so leaving the corresponding constant as experimental

**Review by jsuereth — APPROVED** [review]

File: model/resource/telemetry.yaml (hunk 1 above)
Comment (jsuereth) [reply model/resource/telemetry.yaml]: I think you could stabilize rust here too

**Comment by lmolkova** [comment]
//cc @open-telemetry/semconv-mobile-approvers @open-telemetry/semconv-system-approvers @open-telemetry/semconv-jvm-approvers @open-telemetry/semconv-k8s-approvers @open-telemetry/semconv-container-approvers @open-telemetry/semconv-http-approvers @open-telemetry/semconv-dotnet-approver 

could you please review the yaml file changes for your area? Thanks!

**Review by pyohannes — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Comment by joaopgrassi** [comment]
Thank you for working on this! It's a big milestone 🚀

**Review by AlexanderWert — APPROVED** [review]

**Comment by lmolkova** [comment]
Dear approvers @open-telemetry/semconv-mobile-approvers @open-telemetry/semconv-system-approvers @open-telemetry/semconv-jvm-approvers @open-telemetry/semconv-k8s-approvers @open-telemetry/semconv-container-approvers @open-telemetry/semconv-http-approvers @open-telemetry/semconv-dotnet-approver 

we still want you to check changes in the yaml files in this PR (ms is autogenerated). Please take a look!

**Review by trisch-me — APPROVED** [review]
this is awesome!

**Review by ChrsMark — APPROVED** [review]
LGTM

**Comment by lmolkova** [comment]
@jack-berg @trask @jonatan-ivanov @mateuszrzeszutek could you please check JVM-related yaml changes in this PR?

**Comment by lmolkova** [comment]
@JamesNK @noahfalk @antonfirsov could you please check dotnet-related yaml changes in this PR?

**Comment by lmolkova** [comment]
@nachoBonafonte @LikeTheSalad @bryce-b @breedx-splk @surbhiia could you please check mobile-related yaml changes in this PR?

**Comment by joaopgrassi** [comment]
Sorry for the conflicts, I somehow forgot we were waiting for this 🙈. Got a bit too excited in moving things to the registry 😅

**Review by jack-berg — APPROVED** [review]
JVM stuff looks good to me.

**Review by noahfalk — APPROVED** [review]
dotnet-aspnetcore.yaml, dotnet-kestrel.yaml, and dotnet-signalr.yaml all look fine to me. Were there any other files that are applicable to dotnet?

**Comment by lmolkova** [comment]
@noahfalk this should be it. Thank you!

**Comment by breedx-splk** [comment]
`model/logs/mobile-events.yaml` and `mobile/events.md` both look good to me. 👍🏻

**Pull request merged**